### DISCUSSION TITLE
Secrets as a text file

### DISCUSSION THREAD

**Discussion by travisjohnsonga** [nautobot/nautobot#4655] [General]
This may sound silly - 

But what is an example text file for secrets?

I'm working through an issue with environment variables, so trying to do this as a workaround.

**Comment by glennmatthews** [comment]
Just a plaintext file whose contents are the secret value; e.g. `echo -n mysecretpassword > /tmp/password.txt` then reference `/tmp/password.txt` as the secret.

**Reply by pugnacity** [reply]
Is where any way to define multiple entries per file or is it always one file, one secret?

**Reply by glennmatthews** [reply]
Currently one file, one secret.

Supporting an alternate text file format would not be too difficult to implement if you'd like to give it a try - take a look at some of the secrets providers in https://github.com/nautobot/nautobot-plugin-secrets-providers/ (though those are mostly API/SDK based) to see how it's done; an enhanced text-file format could be a candidate for either that app or for a pull request into core if you get it working. :-)
